aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -121,56 +115,108 @@ void raw_cudnn_convolution_forward_out(
+  auto op = cudnn_frontend::OperationBuilder(desc)
+      .setxDesc(getTensorDescriptor(x, 'x', key.input_alignment))
+      .setyDesc(getTensorDescriptor(y, 'y', key.output_alignment))
+      .setwDesc(getTensorDescriptor(w, 'w', key.weight_alignment))
       .setcDesc(getConvDescriptor(key.params.dataType, padding, stride, dilation))
       .build();
-  // std::cout << op.describe() << std::endl;
 
   std::array<cudnn_frontend::Operation const *, 1> ops = {&op};
 
   auto opGraph = cudnn_frontend::OperationGraphBuilder()
       .setHandle(handle)
       .setOperationGraph(1, ops.data())
       .build();
-  // std::cout << opGraph.describe() << std::endl;
-
   auto heuristics = cudnn_frontend::EngineHeuristicsBuilder()
       .setOperationGraph(opGraph)
       .setHeurMode(CUDNN_HEUR_MODE_INSTANT)
       .build();
   auto fallback = cudnn_frontend::EngineFallbackListBuilder()
                     .setOperationGraph(opGraph)
-                    .setOperation(CUDNN_BACKEND_OPERATION_CONVOLUTION_FORWARD_DESCRIPTOR)
+                    .setOperation(desc)
                     .build();
-
   auto& engine_configs = heuristics.getEngineConfig(heuristics.getEngineConfigCount());
   auto& fallback_list = fallback.getFallbackList();
 
-  cudnn_frontend::EngineConfigList filtered_configs;
-  filterEngineConfigs(engine_configs, filtered_configs, deterministic, allow_tf32, input.scalar_type());
-  filterEngineConfigs(fallback_list, filtered_configs, deterministic, allow_tf32, input.scalar_type());
+  // TODO: Is this ok or do we need to change type deduction based on descriptor?
+  filterEngineConfigs(engine_configs, filtered_configs, deterministic, allow_tf32, x.scalar_type());
+  filterEngineConfigs(fallback_list, filtered_configs, deterministic, allow_tf32, x.scalar_type());
+}
 
+void try_filtered_configs(const cudnn_frontend::EngineConfigList filtered_configs, const CacheKey key, const cudnnHandle_t handle, const Tensor& x, const Tensor& y, const Tensor& w) {
   for (auto &cfg : filtered_configs) {
     try {
-      run(cfg);
+      run_conv_cfg(handle, x, y, w, cfg);
       engine_cache[key] = cfg;
       return;
     } catch (cudnn_frontend::cudnnException &e) {} catch(CuDNNError &e) {}
   }
   TORCH_CHECK(false, "Unable to find an engine to execute this computation");
 }
 
+void run_single_conv(const cudnnBackendDescriptorType_t operation,
+  const Tensor& x, const Tensor& y, const Tensor& w,
+  IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
+  bool benchmark, bool deterministic, bool allow_tf32) {
+  TORCH_CHECK(!benchmark, "not supported yet");
+  if (operation == CUDNN_BACKEND_OPERATION_CONVOLUTION_FORWARD_DESCRIPTOR && y.numel() == 0) {
+    return;
+  } else if (operation == CUDNN_BACKEND_OPERATION_CONVOLUTION_BACKWARD_DATA_DESCRIPTOR && x.numel() == 0) {
+    return;
+  } else if (operation == CUDNN_BACKEND_OPERATION_CONVOLUTION_BACKWARD_FILTER_DESCRIPTOR && w.numel() == 0) {
+    return;
+  }
```

Comment:
Can we move this to `raw_cudnn_convolution_forward_out`, `raw_cudnn_convolution_backward_input_out` and `raw_cudnn_convolution_backward_weight_out `?